mesonbuild/mcompile.py

```diff
@@ -27,9 +29,45 @@
 if T.TYPE_CHECKING:
     import argparse
 
+def generate_ninja_target_name(target_name: str, builddir: Path, introspect_data: dict) -> str:
+    for target in introspect_data:
+        if (not target['subproject'] and target['name'] == target_name):
```

Comment:
please don't wrap if statements in parens unless they span multiple lines, this is python not C :)